tools/benchmark/src/MochaReporter.ts

```diff
@@ -66,7 +66,10 @@ module.exports = class {
 				if (test.state !== "passed") {
 					// The mocha test failed after reporting benchmark data.
 					// This may indicate the benchmark did not measure what was intended, so mark as aborted.
-					const error = `Test ${test.title} in ${suite} completed with status '${test.state}' after reporting data.`;
+					const error =
+						(benchmark as BenchmarkError).error === undefined
```

Comment:
For this line, I believe typecasting to `BenchmarkError` to run a check on `error` is inevitable.

Reply:
Nit suggestion, to avoid having to cast twice:
```
const error = (benchmark as BenchmarkError).error ?? `Test ${test.title} in ${suite} completed with status '${test.state}' after reporting data.`;
```